### PR TITLE
Improve button editbox event

### LINKED ISSUE #619: Button and Editbox add sender arg to Callback

### PR DESCRIPTION
**Pull request by zilongshanren**
fixed https://github.com/cocos-creator/engine/issues/619

Changes proposed in this pull request:
- add sender arg to Button Click event and EditBox event handler. 

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/components/CCEditBox.js [hunk 1]
```diff
@@ -445,15 +445,15 @@ var EditBox = cc.Class({
...
     },
 
     editBoxEditingDidEnded: function() {
-        cc.Component.EventHandler.emitEvents(this.editingDidEnded);
+        cc.Component.EventHandler.emitEvents(this.editingDidEnded, this);
     },
 
     editBoxTextChanged: function(editBox, text) {
-        cc.Component.EventHandler.emitEvents(this.textChanged, text);
+        cc.Component.EventHandler.emitEvents(this.textChanged, text, this);
```
Comment (zilongshanren) [inline cocos2d/core/components/CCEditBox.js:456]: @jareguo  Is it necessary to put `this` before `text`, but it might break back compatibility.

**Comment by zilongshanren** [comment]
The example-case also got updated. https://github.com/cocos-creator/example-cases/pull/124

@nantas  This feature has been requested from many developers

http://forum.cocos.com/t/button/36050

File: cocos2d/core/components/CCComponentEventHandler.js [hunk 2]
```diff
@@ -80,15 +80,14 @@ cc.Component.EventHandler = cc.Class({
         /**
          * @method emitEvents
          * @param {Component.EventHandler[]} events
-         * @param {*} params
          * @statics
          */
-        emitEvents: function(events, params) {
+        emitEvents: function(events) {
```
Comment (jareguo) [inline cocos2d/core/components/CCComponentEventHandler.js:85]: 如果用了 arguments + parameter，麻烦下面加上 `'use strict';` 对 vm 比较友好

File: cocos2d/core/components/CCButton.js [hunk 3]
```diff
@@ -393,7 +393,7 @@ var Button = cc.Class({
         if (!this.interactable || !this.enabledInHierarchy) return;
 
         if (this._pressed) {
-            cc.Component.EventHandler.emitEvents(this.clickEvents, event);
+            cc.Component.EventHandler.emitEvents(this.clickEvents, this, event);
```
Comment (jareguo) [inline cocos2d/core/components/CCButton.js]: 回调的第一个参数变成 target 了？感觉不太对劲，target 应该放到 event.target 提供才对吧？

Comment (zilongshanren) [inline cocos2d/core/components/CCButton.js]: event.target 里面有 不过是 node， node 取出 button，还需要 button.getComponent...

Comment (zilongshanren) [inline cocos2d/core/components/CCButton.js]: @jareguo  `var button = event.target.getComponent(cc.Button)` 要这样

File: cocos2d/core/components/CCComponentEventHandler.js (hunk 2 above)
Comment (zilongshanren) [inline cocos2d/core/components/CCComponentEventHandler.js:85]: @jareguo 在哪里加？

**Pull request merged**